java-frontend/src/main/java/org/sonar/java/se/checks/SECheck.java

```diff
@@ -94,17 +94,18 @@ public void reportIssue(Tree tree, String message, Set<List<JavaFileScannerConte
     Symbol lastEvaluated = currentNode.programState.getLastEvaluated();
     while (node != null) {
       ExplodedGraph.Node finalNode = node;
+      node = node.parent;
```

Comment:
I would have let that line at the end of the while() block, as you are not using `node` anymore before reassigning it to the parent.